keps/sig-scheduling/5075-dra-consumable-capacity/README.md

```diff
@@ -0,0 +1,1256 @@
+  - [ ] (R) [all GA Endpoints](https://github.com/kubernetes/community/pull/1806) must be hit by [Conformance Tests](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/conformance-tests.md) 
+- [ ] (R) Production readiness review completed
+- [ ] (R) Production readiness review approved
+- [ ] "Implementation History" section is up-to-date for milestone
+- [ ] User-facing documentation has been created in [kubernetes/website], for publication to [kubernetes.io]
+- [ ] Supporting documentation—e.g., additional design documents, links to mailing list discussions/SIG meetings, relevant PRs/issues, release notes
+
+[kubernetes.io]: https://kubernetes.io/
+[kubernetes/enhancements]: https://git.k8s.io/enhancements
+[kubernetes/kubernetes]: https://git.k8s.io/kubernetes
+[kubernetes/website]: https://git.k8s.io/website
+
+## Summary
+
+This KEP is an extended usecase from the partitionable device [KEP-4815: DRA: Add support for partitionable devices](https://github.com/kubernetes/enhancements/tree/master/keps/sig-node/4815-dra-partitionable-devices).
+The enhancement enables a shared device allocation with consumable capacity values.
+A shared device can be allocated to more than one resource claims with pre-device resource requests.
+
+## Motivation
+
+A motivating use case for supporting a shared network device which can be selected by more than one pods on demand (on claim). 
+The original discussion is in [this PR's comment thread](https://github.com/kubernetes-sigs/cni-dra-driver/pull/1#discussion_r1889265214).
+The limitation of current implementation has been addressed [here](https://github.com/kubernetes-sigs/cni-dra-driver/pull/1#discussion_r1890166449).
+The virtual network device is created and configured once the CNI is called based on the information of the master network device.
+The configured information specific to the generated device should be at the ResourceClaim, not the ResourceSlice. 
+However, the device in the ResourceClaim is now present, even though the device is listed in the ResourceSlice, and there is no attribute to differentiate between the master device and the actual share of the device.
+
+There are also similar use cases for requesting the device based on a specific root or device type in other contexts,
+including [Peer Pods](https://github.com/confidential-containers/cloud-api-adaptor/blob/main/docs/architecture.md)
+and [Composable Disaggregated Infrastructure (CDI)](https://www.wwt.com/article/an-introduction-to-composable-disaggregated-infrastructure).
+
+This feature is also beneficial for the other sharable devices those are not with a scope of [KEP-4815](https://github.com/kubernetes/enhancements/issues/4815).
+For instance, this feature will be allow reserving memory fraction of virtual GPU in [the AWS virtual GPU device plugin](https://github.com/awslabs/aws-virtual-gpu-device-plugin) via DRA.
+The number of shares can be limited if needed.
+
+### Goals
```

Comment:
So I think the main issue with the partitionable devices feature that this attempts to solve, is to make it possible to share capacity from a device without having to explicitly list every possible partition. We already know this can be very verbose in some situations (example in https://docs.google.com/document/d/1lXGfnrBixRIMW9ESa-mv09Kisb2myVFV_A3nqPJ4FCQ/edit?tab=t.0#bookmark=id.mmv8k6mzxscm). I think it might be useful to spell out more clearly here how this extends the already planned support for partitionable devices (which already allow for sharing devices and to some extent also on-demand sharing).